app/validators/password_validator.rb

```diff
@@ -0,0 +1,31 @@
+
+class PasswordValidator < ActiveModel::EachValidator
+  MIN_LENGTH = 8
+  SPECIAL_CHAR_REGEX = /[!@#$%^&*(),.?":{}|<>]/
+  DIGIT_REGEX = /\d/
+  UPPERCASE_REGEX = /[A-Z]/
+  LOWERCASE_REGEX = /[a-z]/
+
+  def validate_each(record, attribute, value)
+      if value.nil?
+        record.errors.add attribute, :blank_password
+        return
+      end
```

Comment:
bu şekilde yapmak yerine;
```ruby
return record.errors.add(attribute, :blank) if value.blank?
```

şeklinde dersek daha iyi durmaz mı? Hem blank için zaten bir error message'ımız var onuda kullanırız hemde early return vererek daha okunaklı hale getiriirz